match/match.go

```diff
@@ -52,6 +52,36 @@ func jsonMiddleware(next http.Handler) http.Handler {
 	})
 }
 
+func matchCreateHandler(w http.ResponseWriter, r *http.Request) {
+	var req matchDAO.MatchCreateRequest
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		errMsg := utils.CreateErrorJSON(fmt.Sprintf("Invalid request parameters: %s", err.Error()))
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+	if req.UserOne == nil || req.UserTwo == nil {
```

Comment:
Rather than doing this, I was thinking of writing a function that takes any struct and checks if any of the pointers are `nil`. This would make generation easier, but perhaps at the cost of readability.

I looked into this and it seems achievable [using reflection](https://stackoverflow.com/questions/49466522/how-to-check-if-all-fields-of-a-struct-are-nil). I thought I would get others' thoughts before going for it though. Lewis and I suspect Charlie will be all for, but Jay against.

Reply:
I think this is fine

Reply:
https://www.calhoun.io/how-to-determine-if-a-json-key-has-been-set-to-null-or-not-provided/ Could be a solution, although might again be less readable: although, would avoid using reflection ;)